shared/src/main/scala/mlscript/codegen/Codegen.scala

```diff
@@ -869,8 +869,8 @@ final case class JSClassNewDecl(
       implements.foreach { name =>
         buffer += s"    $name.implement(this);"
       }
-      fields.zip(ctorParams).foreach { pair =>
-        buffer += s"    this.#${pair._1} = ${pair._2};" // TODO: invalid name?
+      fields.lazyZip(ctorParams).foreach { (field, param) =>
```

Comment:
Oh and one last thing: it's good practice to always assert that the two zipped collections have the same size, and if it's okay that they do not, document why.

Reply:
Yes they have the same size:
```scala
val ctorParams = fields.map { // ...
```

Have added supplementary comments.

Reply:
Please just assert it and remove the comment that contains the code snippets. Comments like this go out of sync too easily, and the point is that nonlocal invariants like this are easily broken by future careless changes.

Reply:
Oh no I just meant using `assert`. When something should be impossible to happen unless someone mistakenly broke an invariant in a later change, there's no need for a property code generation error!